### PR TITLE
ZEC-23: Set up a database table for products with some initial mock data and unit test.

### PR DESCRIPTION
**Pull request by AnnaMauhay**
ZEC-23: Set up a database table for products with some initial mock data and unit test.

> https://mauhayannaliza.atlassian.net/browse/ZEC-23?atlOrigin=eyJpIjoiOTZiODNmYjA4N2IxNGE4N2IyMjllM2QzYTUzZGNjZmIiLCJwIjoiaiJ9

All unit tests passed.

### CODE REVIEW

File: src/test/java/com/zalando/ecommerce/repository/ProductRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        productList =  productRepository.getProductsByProductNameContainingIgnoreCase("So");
+        assertFalse(productList.isEmpty());
+        assertEquals(1, productList.size());
+
+        productList =  productRepository.getProductsByProductNameContainingIgnoreCase("POO");
+        assertFalse(productList.isEmpty());
+        assertEquals(1, productList.size());
+
+        productList =  productRepository.getProductsByProductNameContainingIgnoreCase("kajsdflkjasd");
+        assertTrue(productList.isEmpty());
```
Comment (MenilV) [inline src/test/java/com/zalando/ecommerce/repository/ProductRepositoryTest.java:78]: I would maybe break this down into multiple tests.

File: src/main/java/com/zalando/ecommerce/controller/UserController.java [hunk 2]
```diff
@@ -33,4 +37,23 @@ public ResponseEntity<?> registerUser(@RequestBody UserRegistrationRequest regis
             return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(e.getMessage());
         }
     }
+
+    @PostMapping(value = "/login")
+    public ResponseEntity<?> login(@RequestBody UserLoginRequest loginRequest){
+        if(!userService.isEmailRegistered(loginRequest.getEmail())) {
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body("The provided email address is not yet registered.");
```
Comment (MenilV) [inline src/main/java/com/zalando/ecommerce/controller/UserController.java:44]: I would hide this for security purposes and just add a generic message so they don't know exactly why login didn't work

**Review by MenilV — COMMENTED** [review]
A few things to check before merging.

File: src/main/java/com/zalando/ecommerce/controller/UserController.java (hunk 2 above)
Comment (AnnaMauhay) [reply src/main/java/com/zalando/ecommerce/controller/UserController.java]: Thanks Menil, I modified it in my new commit.

File: src/test/java/com/zalando/ecommerce/repository/ProductRepositoryTest.java (hunk 1 above)
Comment (AnnaMauhay) [reply src/test/java/com/zalando/ecommerce/repository/ProductRepositoryTest.java]: I've split the test and created a new one that checks an empty list if the search name does not match any of the Product name. Hope this is fine :)

**Review by MenilV — APPROVED** [review]
awesome, approved! 🚀

**Pull request merged**